DigitalLearningSolutions.Web/Controllers/TrackingSystem/CourseSetup/CourseSetupController.cs

```diff
@@ -13,22 +18,99 @@
         }
 
-        public IActionResult Index()
+        [Route("{page=1:int}")]
+        public IActionResult Index(
+            string? searchString = null,
+            string? sortBy = null,
+            string sortDirection = BaseSearchablePageViewModel.Ascending,
+            string? filterBy = null,
+            string? filterValue = null,
+            int page = 1
+        )
         {
+            // Query parameter should take priority over cookie value
+            // We use this method to check for the query parameter rather 
+            // than filterBy != null as filterBy is set to null to clear 
+            // the filter string when javascript is off.
+            if (!Request.Query.ContainsKey(nameof(filterBy)))
+            {
+                filterBy = Request.Cookies[CourseFilterCookieName];
+
+                if (filterBy == null && filterValue == null)
+                {
+                    // Only show Active courses by default
+                    filterValue = CourseStatusFilterOptions.IsActive.FilterValue;
+                }
+            }
+
+            sortBy ??= DefaultSortByOptions.Name.PropertyName;
+            filterBy = FilteringHelper.AddNewFilterToFilterBy(filterBy, filterValue);
+
             var centreId = User.GetCentreId();
             var categoryId = User.GetAdminCategoryId()!;
+            var centreCourses = courseService.GetCentreSpecificCourseStatistics(centreId, categoryId.Value);
+            var categories = courseCategoriesDataService.GetCategoriesForCentreAndCentrallyManagedCourses(centreId).Select(c => c.CategoryName);
+            var topics = courseTopicsDataService.GetCourseTopicsAvailableAtCentre(centreId).Select(c => c.CourseTopic);
 
+            var model = new CourseSetupViewModel(
+                centreCourses,
+                categories,
+                topics,
+                searchString,
+                sortBy,
+                sortDirection,
+                filterBy,
+                page
+            );
+
+            UpdateFilterCookie(filterBy);
+
+            return View(model);
+        }
+
+        [Route("AllCourseStatistics")]
+        public IActionResult AllCourseStatistics()
+        {
+            var centreId = User.GetCentreId();
+            var categoryId = User.GetAdminCategoryId()!;
             var centreCourses = courseService.GetCentreSpecificCourseStatistics(centreId, categoryId.Value);
-            var model = new CourseSetupViewModel(centreCourses.Take(10));
+            var categories = courseCategoriesDataService.GetCategoriesForCentreAndCentrallyManagedCourses(centreId).Select(c => c.CategoryName);
+            var topics = courseTopicsDataService.GetCourseTopicsAvailableAtCentre(centreId).Select(c => c.CourseTopic);
 
+            var model = new AllCourseStatisticsViewModel(centreCourses, categories, topics);
             return View(model);
         }
+
+        private void UpdateFilterCookie(string? filterBy)

```

Comment:
I took a slightly different approach to this comment and only took out the contents of the true branch to a new method. We should be consistent, but I'm not sure which way.

If we take the approach here, the method needs renaming to something referencing the fact that it may also be deleted here.